### PR TITLE
Homework with SpringData

### PR DESCRIPTION
**Pull request by Cherkashyn**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,114 @@
...
+            <version>5.0.3</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (SergeyPasko) [inline pom.xml]: What is it?

File: src/main/java/data/entity/OrderDao.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.math.BigDecimal;
+
+public interface OrderDao {
+
+    Orders findOrderById(BigDecimal id);
+    boolean insertOrder(Orders orders);
+    boolean updateOrder(Orders orders);
+    boolean deleteOrder(BigDecimal id);
+
```
Comment (SergeyPasko) [inline src/main/java/data/entity/OrderDao.java]: You don't need in your task any order logic (in future you must write full tests for it)

File: src/main/java/data/repository/OfficeRepository.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Repository
+public interface OfficeRepository extends JpaRepository<Offices, BigDecimal> {
+
+    @Cacheable("OfficeCache")
```
Comment (SergeyPasko) [inline src/main/java/data/repository/OfficeRepository.java]: Many of them don't have sence - look at the methods in JpaRepository, for example - getAll already present in it.

File: src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+		<SMTP name="FEMAIL"
+			  SMTPProtocol="smtps"
+			  SMTPHost="smtp.gmail.com"
+			  SMTPPort="465"
+			  SMTPUsername="cherkashin17@gmail.com"
+			  SMTPPassword="techinservice"
+			  To="cherkashin17@gmail.com"
+			  from="cherkashin17@gmail.com"
+			  Subject="Calculator Error Log">
```
Comment (SergeyPasko) [inline src/main/resources/log4j2.xml]: Subject="Calculator Error Log". Really?

File: src/test/java/data/CrudOfficeTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
+package data;
```
Comment (SergeyPasko) [inline src/test/java/data/CrudOfficeTest.java]: We will write this in future

File: src/test/java/data/CrudOfficeTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+import org.junit.runners.MethodSorters;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import java.math.BigDecimal;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (SergeyPasko) [inline src/test/java/data/CrudOfficeTest.java]: It is bad appropriach

File: src/main/java/data/repository/OfficeRepository.java (hunk 3 above)
Comment (Cherkashyn) [reply src/main/java/data/repository/OfficeRepository.java]: But I defined them in OfficeService interface. Is this ok to have some methods without interface?

File: src/main/java/officeapp/config/DataConfig.java [hunk 7]
```diff
@@ -22,8 +21,10 @@
 
 @Configuration
 @EnableCaching
+@EnableTransactionManagement
 @EnableJpaRepositories("officeapp.repository")
-@PropertySource("properties/mydb.properties")
+@PropertySource("classpath:officeapp/config/mydb.properties")
+//@Import(MainMvcConfig.class)
```
Comment (Daniluko) [inline src/main/java/officeapp/config/DataConfig.java]: Delete it

File: src/main/java/officeapp/entity/OfficesDao.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package officeapp.entity;
```
Comment (SergeyPasko) [inline src/main/java/officeapp/entity/OfficesDao.java]: What is it?

File: src/test/resources/officeapp/script/schema.sql [hunk 9]
```diff
@@ -0,0 +1,18 @@
+CREATE SCHEMA IF NOT EXISTS MA_STUDENT;
+
+CREATE SEQUENCE IF NOT EXISTS OFFICES_SEQ
```
Comment (SergeyPasko) [inline src/test/resources/officeapp/script/schema.sql]: Where you use it?

**Pull request merged**